homeassistant/components/google_generative_ai_conversation/__init__.py

```diff
@@ -91,8 +95,42 @@ def append_files_to_prompt():
                     )
                     prompt_parts.append(uploaded_file)
 
+        async def wait_for_file_processing(uploaded_file: File) -> None:
+            """Wait for file processing to complete."""
+            uploaded_file = client.files.get(
```

Comment:
To avoid duplicate code you can change this to:

```
while True:
  uploaded_file = client.files.get(...
  if uploaded_file.state not in ...
    break
  await asyncio.sleep...
if uploaded_file.state == FileState.FAILED:
  raise...
```

Reply:
Done